gno.land/pkg/gnoweb/markdown/ext_forms.go

```diff
@@ -0,0 +1,350 @@
+// When a line contains <gno-input />, it adds a child node.
+func (p *formParser) Continue(node ast.Node, reader text.Reader, pc parser.Context) parser.State {
+	line, seg := reader.PeekLine()
+	if seg.Len() == 0 {
+		return parser.Close
+	}
+	trimmed := bytes.TrimSpace(line)
+	// If the line is exactly "</gno-form>", consume it and close the block.
+	if bytes.Equal(trimmed, []byte("</gno-form>")) {
+		reader.Advance(seg.Len())
+		return parser.Close
+	}
+	// If the line starts with "<gno-input", we add an input node.
+	info := parseFormTag(line)
+	if info.Tag == FormTagInput {
+		input := NewForm(FormTagInput)
+
+		// Get realm name and placeholder
+		input.InputName = info.Name
+		input.InputType = info.InputType
+		if info.Placeholder == "" {
+			input.InputPlaceholder = defaultPlaceholder
+		} else {
+			input.InputPlaceholder = info.Placeholder
+		}
+
+		// If an error occurred during parsing, we store it in the node.
+		if info.Error != nil {
+			input.Error = info.Error
+		}
+
+		node.AppendChild(node, input)
+		reader.Advance(seg.Len())
+		return parser.Continue | parser.HasChildren
+	}
+	// Any other line (text, etc.) should stop the block
+	return parser.Close
+}
+
+// Close closes the block
+func (p *formParser) Close(node ast.Node, reader text.Reader, pc parser.Context) {}
+func (p *formParser) CanInterruptParagraph() bool                                { return true }
+func (p *formParser) CanAcceptIndentedLine() bool                                { return true }
+
+// --- The Renderer ---
+
+// formRenderer renders the Form node.
+// When entering the Form node, it displays the opening <form> tag
+// and when exiting (after rendering the child inputs),
+// it displays the submit button and </form>.
+type formRenderer struct{}
+
+// NewFormRenderer creates a new instance of formRenderer
+func NewFormRenderer() *formRenderer {
+	return &formRenderer{}
+}
+
+// RegisterFuncs registers the render function for the Form node
+func (r *formRenderer) RegisterFuncs(reg renderer.NodeRendererFuncRegisterer) {
+	reg.Register(KindForm, r.render)
+}
+
+// render renders the Form node
+func (r *formRenderer) render(w util.BufWriter, source []byte, node ast.Node, entering bool) (ast.WalkStatus, error) {
+	n := node.(*FormNode)
+
+	if n.Tag == FormTagOpen {
+		if entering {
+			// Form action must include the full path
+			formAction := n.RealmName // start with /r/docs/markdown
+			if n.RenderPath != "" {
+				formAction += ":" + strings.TrimPrefix(n.RenderPath, "/")
+			}
+
+			// Render form opening and header
+			fmt.Fprintf(w, `<form class="gno-form" method="post" action="%s">`, formAction)
+			fmt.Fprintln(w, `<div class="gno-form_header">`)
+			fmt.Fprintf(w, `<span><span class="font-bold"> %s </span> Form</span>`, n.RealmName)
+			fmt.Fprintf(w, `<span class="tooltip" data-tooltip="Processed securely by %s."><svg class="w-4 h-4"><use href="#ico-info"></use></svg></span>`, n.RealmName)
```

Comment:
nit, looks nicer imo
```suggestion
			fmt.Fprintf(w, `<span class="tooltip" data-tooltip="Processed securely by %s"><svg class="w-4 h-4"><use href="#ico-info"></use></svg></span>`, n.RealmName)
```